cmd/dep/ensure_test.go

```diff
@@ -186,3 +186,102 @@ func TestDeduceConstraint(t *testing.T) {
 		}
 	}
 }
+
+func TestEnsureUpdate(t *testing.T) {
+	test.NeedsExternalNetwork(t)
+	test.NeedsGit(t)
+
+	h := test.NewHelper(t)
+	defer h.Cleanup()
+
+	h.TempDir("src")
+	h.Setenv("GOPATH", h.Path("."))
+
+	m := `package main
+
+import (
```

Comment:
However you like.  I have my intial conversions parked on a local branch, and I haven't gotten to `ensure_test.go` yet.  You can finish up the file, or leave it to me - I'm good either way.